feather/plugin-host/src/command_dispatcher.rs

```diff
@@ -0,0 +1,163 @@
+use std::{
+    cell::RefCell,
+    convert::TryInto,
+    fmt::{self, Debug},
+    rc::Rc,
+};
+
+use crate::{context::PluginPtrMut, Game, PluginId, PluginManager};
+use anyhow::bail;
+use feather_ecs::{Entity, EntityRef};
+use lieutenant::regex::early_termination::CmdPos;
+use lieutenant::regex::{dfa::DFA, NFA};
+use quill_common::EntityId;
+
+// Constant that should reflect how many commands a default
+// server starts with. Vanilla has roughly 60
+const NUM_COMMANDS: usize = 20;
+
+type CommandId = u32;
+
+/// A collection of data containing what plugin a command was defined in,
+/// the pointer to its Box<Command<...>>, and the regex that determines
+/// when the command should be called.(The command does some parsing beyond the regex).
+pub struct CommandDescription {
+    pub plugin_id: PluginId,
+    pub regex: String,
```

Comment:
Why are we using a regex in here, if we are already using lieutenant?

Reply:
Llieutenant internally uses a custom Regex engine, this just passes the Regex from client(plugin) to host(server) such the server can ask the client(plugin) to do the final matching.

Reply:
Ah thanks for the clarification, i only used regular (java) brigadier :)